### PR TITLE
fix promise

### PR DESCRIPTION
**Pull request by GaoXanSheng**
4f97168fa94027c599a003cded48d854
55391711b74daec94954a3be584026d1
ea4b92b553048df1f29daf2df677b4ce

### CODE REVIEW

**Comment by GaoXanSheng** [comment]
Perhaps I should inherit from HumanoidArmorLayer and HumanoidModelRenderer?
Forget it, as long as it's implemented, it's fine

**Comment by MarkusBordihn** [comment]
Thank you for your PR. Unfortunately, this implementation seems more like a temporary workaround rather than a maintainable long-term solution.

The "drown" model consists of two layers and associated texture files. By merging them into one, you would eliminate support for alternative "drown" textures, requiring users to create custom textures specifically for this mod.

However, the next major update after the full 5.x release will include some fixes to address this issue and other issues.

If you can support the "drown" model as it is, using the original textures, I would be happy to accept this PR. 
Introducing additional hacks is not something I can support long-term. 

Nonetheless, since this is an open-source project, you are free to fork it, provided you do not copy any creative assets from the original version.

**Comment by GaoXanSheng** [comment]
If you want to keep it as it is, the best solution is to render drowned_outer_layer as armor.
So I should let easy_npc_epic_fight handle the image merging internally?

**Comment by MarkusBordihn** [comment]
The solution should function without the need for any custom textures, such as:
Forge/src/main/resources/assets/easy_npc_epic_fight/textures/entity/zombie/drowned.png

Your fix should utilize the vanilla textures to ensure compatibility with existing texture packs. 
Using a custom texture or rendering the drowned_outer_layer as armor would cause other drowned texture packs to malfunction. While this might resolve the specific issues, it would break compatibility with texture packs.

**Comment by GaoXanSheng** [comment]
I know, so I want to combine two textures into one in Minecraft and then render that texture. Alternatively, I could render the file as armor by simply adjusting a simple offset to render it beneath the armor. The biggest problem right now is that I don't have time

**Comment by GaoXanSheng** [comment]
81d519ed-fe1c-410b-bdbf-35a5a89393d1
This is the current performance

File: gradle.properties
Comment (MarkusBordihn) [inline gradle.properties:1]: Please revert this file. The 5.x beta will have broken changes with the next update.
I will make the corresponding updates, as soon 5.x is ready for release.

File: Forge/src/main/resources/META-INF/mods.toml
Comment (MarkusBordihn) [inline Forge/src/main/resources/META-INF/mods.toml:1]: Please revert this file. The restrictions make sure that they are no broken changes and conflicts.

File: Forge/src/main/java/de/markusbordihn/easynpcepicfight/client/renderer/HumanoidModelRenderer.java
Comment (MarkusBordihn) [inline Forge/src/main/java/de/markusbordihn/easynpcepicfight/client/renderer/HumanoidModelRenderer.java:1]: Fyi: Much better. Love this approach will run some test and maybe implement this directly in the core mod.

**Review by MarkusBordihn — APPROVED** [review]
Thanks a lot for the extra time to make the additional changes! 😊

This looks much better from a technical point of view. As soon as you revert the two requested files, I'll merge this PR and start working on releasing the updated versions for 4.x. 🚀

For version 5.x, the update will take a little longer because I need to backport some changes required for Minecraft 1.21, which would break the current implementation. 

Unfortunately, 1.20.6 and 1.21 have many breaking changes that require a lot of additional work and testing. 😔

Thanks for your understanding and patience!

**Comment by GaoXanSheng** [comment]
This is just a solution for the current time period; perhaps in the future, there will be better ones.

**Comment by GaoXanSheng** [comment]
I have an idea, although the current skull looks like 2D, I can have a built-in method to make the drowned corpse into a double-layer skin, keeping the current texture unchanged, and adding 3D effects to the skull

Perhaps it will take some time

**Comment by GaoXanSheng** [comment]
1558a85f-edbb-4e6f-aa6c-4cbf4bd9c7b6

This is the current performance

**Review by MarkusBordihn — APPROVED** [review]

**Comment by MarkusBordihn** [comment]
Take your time and let me know when the code is ready to merge

In general with the 5.x version there will be better native support for Epic Fight, additional to the Fabric and NeoForge support.

**Comment by GaoXanSheng** [comment]
I have fixed the build error and there should be no problem now

**Review by MarkusBordihn — APPROVED** [review]

**Pull request merged**